lib/ace/lib/event.js

```diff
@@ -55,12 +55,24 @@ function getListenerOptions() {
     return activeListenerOptions;
 }
 
-exports.addListener = function(elem, type, callback) {
-    return elem.addEventListener(type, callback, getListenerOptions());
+function EventListener(elem, type, callback) {
+    this.elem = elem;
+    this.type = type;
+    this.callback = callback;
+}
+EventListener.prototype.destroy = function() {
+    removeListener(this.elem, this.type, this.callback);
+    this.elem = this.type = this.callback = undefined;
+};
+
+var addListener = exports.addListener = function(elem, type, callback, destroyable) {
+    elem.addEventListener(type, callback, getListenerOptions());
+    if (destroyable && destroyable.$toDestroy) 
```

Comment:
IfI understand this correctly, all of our eventlisteners are required to clean up after themselves now. Which makes this parameter _strongly_ suggested if not _required_.

Can we make this parameter required? it's easy to unintentionally forget the parameter or supply the wrong destroyable, which would not be detected currently.

If we can't make it required, we can at least enforce that if supplied, it's a proper destroyable and must contain the $toDestroy array.